packages/hydrogen/src/customer/types.ts

```diff
@@ -0,0 +1,139 @@
+  query: <
+    OverrideReturnType extends any = never,
+    RawGqlString extends string = string,
+  >(
+    query: RawGqlString,
+    ...options: ClientVariablesInRestParams<
+      CustomerAccountQueries,
+      RawGqlString
+    >
+  ) => Promise<
+    CustomerAPIResponse<
+      ClientReturn<CustomerAccountQueries, RawGqlString, OverrideReturnType>
+    >
+  >;
+  /** Execute a GraphQL mutation against the Customer Account API. Usually you should first check if the user is logged in before querying the API. */
+  mutate: <
+    OverrideReturnType extends any = never,
+    RawGqlString extends string = string,
+  >(
+    mutation: RawGqlString,
+    ...options: ClientVariablesInRestParams<
+      CustomerAccountMutations,
+      RawGqlString
+    >
+  ) => Promise<
+    CustomerAPIResponse<
+      ClientReturn<CustomerAccountMutations, RawGqlString, OverrideReturnType>
+    >
+  >;
+};
+
+export type CustomerClientOptions = {
+  /** The client requires a session to persist the auth and refresh token. By default Hydrogen ships with cookie session storage, but you can use [another session storage](https://remix.run/docs/en/main/utils/sessions) implementation.  */
+  session: HydrogenSession;
+  /** Unique UUID prefixed with `shp_` associated with the application, this should be visible in the customer account api settings in the Hydrogen admin channel. Mock.shop doesn't automatically supply customerAccountId. Use `h2 env pull` to link your store credentials. */
+  customerAccountId: string;
+  /** The account URL associated with the application, this should be visible in the customer account api settings in the Hydrogen admin channel. Mock.shop doesn't automatically supply customerAccountUrl. Use `h2 env pull` to link your store credentials. */
+  customerAccountUrl: string;
+  /** Override the version of the API */
+  customerApiVersion?: string;
+  /** The object for the current Request. It should be provided by your platform. */
+  request: CrossRuntimeRequest;
+  /** The waitUntil function is used to keep the current request/response lifecycle alive even after a response has been sent. It should be provided by your platform. */
+  waitUntil?: ExecutionContext['waitUntil'];
+  /** This is the route in your app that authorizes the user after logging in. Make sure to call `customer.authorize()` within the loader on this route. It defaults to `/account/authorize`. */
+  authUrl?: string;
+  /** The behaviour when handleUnauthorized, query or mutate is called with an unauthorized customer. The return of method will be throw. The default behaviour is redirecting to `/account/login` where login() is located and pass in current path as redirectPath. */
+  unauthorizedHandler?: () => DataFunctionValue;
+};
+
+/** Below are types meant for documentation only. Ensure it stay in sycn with the type above. */
```

Comment:
Nit, but if the type below is just meant for docs, then probably we only need to keep the comments down there. Less to keep in sync.